### DISCUSSION TITLE
Missing name member in texture objects

### DISCUSSION THREAD

**Discussion by jaspervdv** [cityjson/specs#152] [Q&A]
Currently material objects have a member "name" which can be used to give a short description/...name of the material. Textures do not have this field however. In theory a name can be derived by searching for the name that was given to the texture in a geometry object. But this is quite a process and makes it challenging to work with, especially for a user that is not familiar with the data or well known with the cityjson format.

**Comment by balazsdukai** [comment]
My understanding is that you can use image path in `Texture.image` if you need to store a name for the texture, and thus the texture name is already in the file path. If you don't want extract the filename from the path, you could probably use the full path as the name.

Also, while usually there are a couple materials that are applied across all surfaces, from textures we normally have a different one for each surface. Thus I feel it would be redundant to assign a name besides the file path to a Texture.

**Comment by jaspervdv** [comment]
hemm, yes, I have been pointed to using the filepath as name, that seems to be the most sensible solution currently. I was just surprised by a user using the grasshopper plugin that started using generic tiled textures for roof surfaces. So basically using a texture as a material and assigning it to a lot of surfaces.  However, using the path would be a solution to that problem
